src/utils/resources/vm/utils/network/network-columns.tsx

```diff
@@ -0,0 +1,89 @@
+import { TFunction } from 'react-i18next';
+
+import { NetworkPresentation } from '@kubevirt-utils/resources/vm/utils/network/constants';
+import { getConfigInterfaceState } from '@kubevirt-utils/resources/vm/utils/network/selectors';
+import { sortNICs } from '@kubevirt-utils/resources/vm/utils/network/utils';
+import { sortByDirection, universalComparator } from '@kubevirt-utils/utils/utils';
+import { TableColumn } from '@openshift-console/dynamic-plugin-sdk';
+import { sortable, SortByDirection } from '@patternfly/react-table';
+
+export const compareWithDirection = (direction: SortByDirection, a: any, b: any) =>
+  sortByDirection(universalComparator, direction)(a, b);
+
+export const toNetworkNameLabel = <
+  T extends { network: { multus?: { networkName: string }; pod?: unknown } },
+>(
+  t: TFunction,
+  nic: T,
+) => (nic?.network?.pod ? t('Pod networking') : nic?.network?.multus?.networkName);
+
+export const isSRIOVNetwork = <T extends { sriov?: object }>(iface: T) => !!iface?.sriov;
+
+export const Name = <T extends { network: { name: string } }>(t: TFunction): TableColumn<T> => ({
```

Comment:
Maybe I'm missing something, but is there a reason we're not just using `NetworkPresentation` here?

Reply:
why moving t around? get directly

Reply:
@metalice 
1. regarding changing the location - the main motivation is to re-use column definition - before this PR there were 3 places with (almost) the same code
2. regarding changing the way we build column definition:
      1. the previous way of getting the value for sorting (as string) was prone to bugs (i.e. [sorting by network](https://github.com/kubevirt-ui/kubevirt-plugin/blob/976597487d3653dbbfd6f630f24fb03b8690fc27/src/views/virtualmachines/details/tabs/configuration/network/hooks/useNetworkColumns.ts#L30)). With string typing the props will be in sync.
      2. column builder as function allows to pass translations in a proper way
4. regarding the use of generics - with changes in #2662 the network columns in VM details will have different entity then in catalog